torchrl/data/replay_buffers/replay_buffers.py

```diff
@@ -90,12 +95,17 @@ class ReplayBuffer:
             samples.
         prefetch (int, optional): number of next batches to be prefetched
             using multithreading. Defaults to None (no prefetching).
-        transform (Transform, optional): Transform to be executed when
+        transform (Transform or Callable[[Any], Any], optional): Transform to be executed when
             :meth:`sample` is called.
             To chain transforms use the :class:`~torchrl.envs.Compose` class.
             Transforms should be used with :class:`tensordict.TensorDict`
             content. A generic callable can also be passed if the replay buffer
             is used with PyTree structures (see example below).
+            Unlike storages, writers and samplers, transform constructors must
+            be passed as separate keyword argument :attr:`transform_factory`,
+            as it is impossible to distinguish a constructor from a transform.
+        transform_factory (Callable[[], Callable], optional): a factory for the
```

Comment:
If we do this here for all of the buffers it kind of feels like the API for collector should match 😄

Reply:
Yeah I agree
The more I think about it the more it's obvious we should just pass a factory to the collectors